nomad/job_endpoint_hook_implicit_identities.go

```diff
@@ -74,18 +74,20 @@ func (h jobImplicitIdentitiesHook) handleConsulService(s *structs.Service) {
 	}
 
 	// Set the expected identity name and service name.
-	serviceWID.Name = fmt.Sprintf("%s/%s", consulServiceIdentityNamePrefix, s.Name)
+	name := fmt.Sprintf("_nomad-service-%v-%v-%v", s.TaskName, s.Name, s.PortLabel)
```

Comment:
Can we turn this into a function that can be imported on the client as well (so probably in `nomad/structs`), so that the downstream hooks can use the same function and we guarantee they're all trying to get the same service WID when we're done? 

Also, given that we're prefixing this with `consul-service` does it need to also start with `_nomad-service`? The long prefix `consul-service/_nomad-service-taskName-serviceName-port` seems redundant.

Reply:
Have a look at ec96df1ce21f4e0b0b65b8c060affdc3606a21fa, I think it's what you suggested.

Reply:
Perfect :+1: